src/checkers/inference/InferenceAnnotatedTypeFactory.java

```diff
@@ -363,6 +363,20 @@ public Boolean visitDeclared(AnnotatedDeclaredType type, Void p) {
         }
     };
 
+    /**
+     * Override this method to re-add EnclosingType type into AnnotatedDeclaredType of a class.
+     */
+    @Override
+    @SuppressWarnings("deprecation")
```

Comment:
Now that https://github.com/eisop/checker-framework/pull/267 is merged, do we need this part? Can you go over this whole PR and see which parts we still need? (At least keeping the test case will be good!)

Reply:
I'll merge #395 first, as all JDK 8 and 11 downstream tests pass now.
Removing the usage of the deprecated method and the test case would still be good to have.
Please re-open this PR against master and minimize it as necessary.

Reply:
Thanks! Let me re-oopen it.